test/e2e/network/netpol/k8s_util.go

```diff
@@ -0,0 +1,373 @@
+		l, err := k.ClientSet.NetworkingV1().NetworkPolicies(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "unable to list network policies in ns %s", ns)
+		}
+		for _, np := range l.Items {
+			log.Infof("deleting network policy %s in ns %s", np.Name, ns)
+			err = k.ClientSet.NetworkingV1().NetworkPolicies(ns).Delete(context.TODO(), np.Name, metav1.DeleteOptions{})
+			if err != nil {
+				return errors.Wrapf(err, "unable to delete network policy %s", np.Name)
+			}
+		}
+	}
+	return nil
+}
+
+// ClearCache clears the kube pod cache
+func (k *Kubernetes) ClearCache() {
+	log.Info("Clearing pod cache")
+	k.mutex.Lock()
+	k.podCache = map[string][]v1.Pod{}
+	k.mutex.Unlock()
+	log.Info("Pod cache successfully cleared")
+}
+
+// CreateOrUpdateNetworkPolicy is a convenience function for updating/creating netpols
+func (k *Kubernetes) CreateOrUpdateNetworkPolicy(ns string, netpol *networkingv1.NetworkPolicy) (*networkingv1.NetworkPolicy, error) {
+	log.Infof("creating/updating network policy %s in ns %s", netpol.Name, ns)
+	netpol.ObjectMeta.Namespace = ns
+	np, err := k.ClientSet.NetworkingV1().NetworkPolicies(ns).Update(context.TODO(), netpol, metav1.UpdateOptions{})
+	if err == nil {
+		return np, err
+	}
+
+	log.Debugf("unable to update network policy %s in ns %s, let's try creating it instead (error: %s)", netpol.Name, ns, err)
+	np, err = k.ClientSet.NetworkingV1().NetworkPolicies(ns).Create(context.TODO(), netpol, metav1.CreateOptions{})
+	if err != nil {
+		log.Debugf("unable to create network policy: %s", err)
+	}
+	return np, err
+}
+
+func (k *Kubernetes) waitForHTTPServers(model *Model) error {
+	const maxTries = 10
+	const sleepInterval = 1 * time.Second
+	log.Infof("waiting for HTTP servers (ports 80 and 81) to become ready")
+
+	testCases := map[string]*TestCase{}
+	for _, port := range model.Ports {
+		for _, protocol := range model.Protocols {
+			fromPort := 81
+			desc := fmt.Sprintf("%d->%d,%s", fromPort, port, protocol)
+			testCases[desc] = &TestCase{FromPort: fromPort, ToPort: int(port), Protocol: protocol}
+		}
+	}
+	notReady := map[string]bool{}
+	for caseName := range testCases {
+		notReady[caseName] = true
+	}
+
+	for i := 0; i < maxTries; i++ {
+		for caseName, testCase := range testCases {
+			if notReady[caseName] {
+				reachability := NewReachability(model.AllPods(), true)
+				testCase.Reachability = reachability
+				ProbePodToPodConnectivity(k, model, testCase)
+				_, wrong, _ := reachability.Summary()
+				if wrong == 0 {
+					log.Infof("server %s is ready", caseName)
+					delete(notReady, caseName)
+				} else {
+					log.Infof("server %s is not ready", caseName)
+				}
+			}
+		}
+		if len(notReady) == 0 {
+			return nil
+		}
+		time.Sleep(sleepInterval)
+	}
+	return errors.Errorf("after %d tries, %d HTTP servers are not ready", maxTries, len(notReady))
+}
+
+func (k *Kubernetes) waitForPodInNamespace(ns string, pod string) error {
```

Comment:
duplicate of 
https://github.com/kubernetes/kubernetes/blob/ac10e4043af29c651407add4ead61975471f2fbd/test/e2e/framework/pod/wait.go#L315-L319

Reply:
👍